### PR TITLE
Okta IAM - phase 2

### PR DESCRIPTION
**Pull request by dantavori**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/30103

## Description
Okta IAM: added `fetch-incidents`, `okta-list-applications` and `okta-get-app-user-assignment ` commands

for fetch incidents API request, we have two ways to get the query filter string: either directly from the fetch_query_filter param, or automatically generate it from the IAM configuration.

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [x] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/Okta/Integrations/Okta_IAM/Okta_IAM.py [hunk 1]
```diff
@@ -362,6 +411,63 @@ def update_user_command(client, args, mapper_out, is_command_enabled, is_create_
     return user_profile
 
 
+def get_assigned_user_for_app_command(client, args):
+    user_id = args.get('user-id')
```
Comment (anara123) [inline Packs/Okta/Integrations/Okta_IAM/Okta_IAM.py:415]: name of args `user_id`

File: Packs/Okta/Integrations/Okta_IAM/Okta_IAM.py [hunk 2]
```diff
@@ -427,6 +537,12 @@ def main():
         elif command == 'get-mapping-fields':
             return_results(get_mapping_fields_command(client))
 
+        elif command == 'okta-get-assigned-user-for-app':
+            return_results(get_assigned_user_for_app_command(client, args))
+
+        elif command == 'fetch-incidents':
+            demisto.incidents(fetch_incidents(client, fetch_query_filter, fetch_time, fetch_limit))
```
Comment (anara123) [inline Packs/Okta/Integrations/Okta_IAM/Okta_IAM.py:544]: fetch_incidents should return `next_run` and `incidents`

File: Packs/Okta/Integrations/Okta_IAM/Okta_IAM.py [hunk 3]
```diff
@@ -362,6 +411,63 @@ def update_user_command(client, args, mapper_out, is_command_enabled, is_create_
...
+    last_run_time = last_run.get('last_run_time', fetch_time)  # if last_run_time is undefined, use fetch_time param
+    time_now = datetime.now().strftime(DATE_FORMAT)
+
+    demisto.debug(f'Okta: Fetching logs from {last_run_time} to {time_now}.')
+    if not incidents:
+        try:
+            log_events = client.get_logs(query_filter, last_run_time, time_now)
+            for entry in log_events:
+                # mapping will be done at the classification and mapping
+                incident = {'rawJSON': json.dumps(entry)}
```
Comment (anara123) [inline Packs/Okta/Integrations/Okta_IAM/Okta_IAM.py:456]: add `'name'` field to the incident

File: Packs/Okta/Integrations/Okta_IAM/Okta_IAM.py [hunk 4]
```diff
@@ -362,6 +411,63 @@ def update_user_command(client, args, mapper_out, is_command_enabled, is_create_
...
+                incidents.append(incident)
+
+        except Exception as e:
+            demisto.error(f'Failed to fetch Okta log events from {last_run_time} to {time_now}.')
+            raise e
+
```
Comment (anara123) [inline Packs/Okta/Integrations/Okta_IAM/Okta_IAM.py:462]: remove try and catch

File: Packs/Okta/Integrations/Okta_IAM/README.md [hunk 5]
```diff
@@ -19,8 +19,15 @@ For more information, please refer to the [Identity Lifecycle Management article
...
 | update-user-enabled | Update User Command Enabled | False |
 | enable-disable-user-enabled | Enable/Disable User Commands Enabled | False |
+| create-if-not-exists | Automatically create user if not found in update and enable commands | False |
 | mapper-in | Incoming Mapper | True |
 | mapper-out | Outgoing Mapper | True |
+| max_fetch | Fetch Limit \(recommended less than 200\) | False |
+| isFetch | Fetch incidents | False |
+| incidentFetchInterval | Incidents Fetch Interval | False |
+| incidentType | Incident type | False |
+| fetch_query_filter | Fetch Query Filter | True |
```
Comment (anara123) [inline Packs/Okta/Integrations/Okta_IAM/README.md:29]: Provide more documentation, maybe link to the query syntax page. Where the user should query for these logs.
Maybe example

**Review by anara123 — CHANGES_REQUESTED** [review]

File: Packs/Okta/Integrations/Okta_IAM/Okta_IAM.py [hunk 6]
```diff
@@ -111,26 +113,129 @@ def get_okta_fields(self):
...
+            url_suffix=uri,
+            resp_type='response',
+            ok_codes=(200, 404)
+        )
+        return res.status_code == 200  # True if user is assigned to app, false otherwise
+
+    def list_apps(self, query, page, limit):
+        query_params = {
+            'q': encode_string_results(query),
+            'limit': encode_string_results(limit)
```
Comment (anara123) [inline Packs/Okta/Integrations/Okta_IAM/Okta_IAM.py:129]: why do you use encode_string_results for limit ? It should be a number, no?

File: Packs/Okta/Integrations/Okta_IAM/Okta_IAM.py [hunk 7]
```diff
@@ -111,26 +113,129 @@ def get_okta_fields(self):
...
+        }
+
+        curr_page = 0
+        apps_batch, next_page = self.list_apps_batch(url_suffix='/apps', params=query_params)
```
Comment (anara123) [inline Packs/Okta/Integrations/Okta_IAM/Okta_IAM.py:133]: if you don't pass the page arg to the query, it means the API has its own page size? you do while till you reach the limit?

**Review by anara123 — APPROVED** [review]
Feel free to merge when you are ready

**Comment by yaron-libman** [comment]
Reviewed.

**Pull request merged**